src/neo4j-schema/types.js

```diff
@@ -0,0 +1,106 @@
+import _ from 'lodash';
+
+/**
+ * Choose a single property type given an array of possible values.  In the simplest
+ * and usual case, a property only has one possible type, and so it will get assigned
+ * that.  But we have to handle situations where a property can be a long or an int,
+ * depending on value, etc.
+ * @param {Object} property a property object from OKAPI schema information
+ * @returns String a single type name.
+ */
+const chooseGraphQLType = property => {
+  const options = _.get(property, 'propertyTypes');
+  const mandatoryModifier = _.get(property, 'mandatory') ? '!' : '';
+
+  // This stores and performs the mapping of a primitive Neo4j type t
+  // to a GraphQL primitive type (the output)
+  const mapSingleType = t => {
+    const mapping = {
+      // Primitives
+      Long: 'Int',
+      Float: 'Float',
+      Double: 'Float',
+      String: 'String',
+      Boolean: 'Boolean',
+      Date: 'Date',
+      DateTime: 'DateTime',
+      LocalTime: 'LocalTime',
+      LocalDateTime: 'LocalDateTime',
+      Time: 'Time',
+
+      // Array types
+      LongArray: '[Int]',
+      DoubleArray: '[Float]',
+      FloatArray: '[Float]',
+      BooleanArray: '[Boolean]',
+      StringArray: '[String]',
+      DateArray: '[Date]',
+      DateTimeArray: '[DateTime]',
+      TimeArray: '[Time]',
+      LocalTimeArray: '[LocalTime]',
+      LocalDateTimeArray: '[LocalDateTime]'
+    };
+
+    return mapping[t] || 'String';
```

Comment:
I tried to be exhaustive on mapping potential neo4j types, but yeah I think I missed "Point".

I don't think we can get away with omitting "Points" though, becuase the auto-gen schema is going to get fed to augment, which will accept the omission I guess.  Maybe we need a carve out and to omit those properties pending point support, but ultimately I guess I need some directive, or other way to express that it's a point so that augment can do the right thing.  What do you think?

Reply:
Let's leave this as-is then - with String as the default. And I can add Point to the inferSchema mapping once we support it in neo4j-graphql-js.  I think Point is the only complex type not supported by neo4-graphql-js and we plan to add it very soon. So in the meantime Point fields will be treated as strings, I'm OK with that.